nft-minter/src/nft_module.rs

```diff
@@ -0,0 +1,211 @@
+                self.nft_token(&brand_id).set_token_id(&token_id);
+            }
+            ManagedAsyncCallResult::Err(_) => {
+                self.brand_info(&brand_id).clear();
+                self.tags_for_brand(&brand_id).clear();
+                self.available_ids(&brand_id).clear_len();
+                let _ = self.registered_brands().swap_remove(&brand_id);
+            }
+        }
+    }
+
+    #[endpoint(setLocalRoles)]
+    fn set_local_roles(&self, brand_id: BrandId<Self::Api>) {
+        self.nft_token(&brand_id)
+            .set_local_roles(&[EsdtLocalRole::NftCreate], None);
+    }
+
+    #[payable("*")]
+    #[endpoint(buyRandomNft)]
+    fn buy_random_nft(&self, brand_id: BrandId<Self::Api>) {
+        require!(
+            self.registered_brands().contains(&brand_id),
+            INVALID_BRAND_ID_ERR_MSG
+        );
+
+        let brand_info: BrandInfo<Self::Api> = self.brand_info(&brand_id).get();
+        let payment: EsdtTokenPayment<Self::Api> = self.call_value().payment();
+        require!(
+            payment.token_identifier == brand_info.mint_price_token_id
+                && payment.amount == brand_info.mint_price_amount,
+            "Invalid payment"
+        );
+
+        self.add_mint_payment(payment.token_identifier, payment.amount);
+
+        let caller = self.blockchain().get_caller();
+        self.send_random_nft(&caller, &brand_id, &brand_info);
+    }
+
+    #[endpoint(givawayNft)]
+    fn giveaway_nft(&self, to: ManagedAddress, brand_id: BrandId<Self::Api>) {
+        self.require_caller_is_admin();
+
+        let brand_info = self.brand_info(&brand_id).get();
+        self.send_random_nft(&to, &brand_id, &brand_info);
+    }
+
+    fn send_random_nft(
+        &self,
+        to: &ManagedAddress,
+        brand_id: &BrandId<Self::Api>,
+        brand_info: &BrandInfo<Self::Api>,
+    ) {
+        let nft_id = self.get_next_random_id(brand_id, brand_info.id_offset);
+        let nft_uri = self.build_nft_main_file_uri(nft_id, &brand_info.media_type);
+        let nft_json = self.build_nft_json_file_uri(nft_id);
+        let collection_json = self.build_collection_json_file_uri();
+
+        let mut uris = ManagedVec::new();
+        uris.push(nft_uri);
+        uris.push(nft_json);
+        uris.push(collection_json);
+
+        let attributes = self.build_nft_attributes(brand_id, nft_id);
+        let nft_token_id = self.nft_token(brand_id).get_token_id();
+        let nft_amount = BigUint::from(NFT_AMOUNT);
+        let nft_nonce = self.send().esdt_nft_create(
+            &nft_token_id,
+            &nft_amount,
+            &brand_info.token_display_name,
+            &brand_info.royalties,
+            &ManagedBuffer::new(),
+            &attributes,
+            &uris,
+        );
+
+        self.send()
+            .direct(to, &nft_token_id, nft_nonce, &nft_amount, &[]);
+    }
+
+    fn get_next_random_id(&self, brand_id: &BrandId<Self::Api>, id_offset: usize) -> UniqueId {
```

Comment:
What happens here if the id was prev random generated?

Reply:
Can't happen, the ID is removed from the mapper once used.